google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -549,13 +556,37 @@ public Integer getDaysSinceNoncurrentTime() {
 
       /**
        * Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").
+       *
+       * @deprecated {@link #getNoncurrentTimeBeforeOffsetDateTime()}
        */
+      @Deprecated
       public DateTime getNoncurrentTimeBefore() {
+        if (noncurrentTimeBefore == null) return null;
+        return new DateTime(millisOffsetDateTimeCodec.decode(noncurrentTimeBefore));
+      }
+
+      /**
+       * Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").
```

Comment:
`OffesetDateTime` isn't exactly an RFC 3339 date.

I think we'll want to change the comment here to be a bit more explicit. Something like:

> Returns the date and offset from UTC for this condition.
> If a time other than `00:00:00.000` is present in the value, GCS will truncate to `00:00:00.000`.